gordo/machine/dataset/datasets.py

```diff
@@ -253,6 +259,13 @@ def get_data(self) -> Tuple[pd.DataFrame, Optional[pd.DataFrame]]:
                     f"specified required threshold for number of rows ({self.n_samples_threshold})."
                 )
 
+        if self.filter_periods:
+            filtered_periods = filter_periods(
+                data=data, granularity=self.resolution, kwargs=self.filter_periods
+            )
+            data = filtered_periods.data
+            self._metadata["filtered_periods"] = filtered_periods.drop_periods
```

Comment:
As I will suggest on the bottom of this page `FilteredPeriods` should have `filter_df` method
```suggestion
        if self.filter_periods:
            data, drop_periods = self.filter_periods.filter_df(data)
            self._metadata["filtered_periods"] =drop_periods
```